src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -233,36 +248,56 @@ void runInternal() throws IOException, BackoffException {
 
         while (true) {
             try {
-
                 LOG.debug("Waiting for next batch of events for [{}]", eventName);
 
                 expectToken(jsonParser, JsonToken.START_OBJECT);
-                expectToken(jsonParser, JsonToken.FIELD_NAME);
-                expectField(jsonParser, "cursor");
-
-                final Cursor cursor = readCursor(jsonParser);
+                metricsCollector.markMessageReceived();
 
-                LOG.debug("Cursor for [{}] partition [{}] at offset [{}]", eventName, cursor.getPartition(), cursor.getOffset());
+                Cursor cursor = null;
+                List<T> events = null;
+
+                while (jsonParser.nextToken() != JsonToken.END_OBJECT) {
+                    final String field = jsonParser.getCurrentName();
+                    switch (field) {
+                        case "cursor": {
+                            cursor = readCursor(jsonParser);
+                            break;
+                        }
+                        case "events": {
+                            events = readEvents(jsonParser);
+                            break;
+                        }
+                        case "metadata": {
+                            LOG.debug("Skipping metadata in event batch");
+                            jsonParser.nextToken();
+                            jsonParser.skipChildren();
+                            break;
+                        }
+                        default: {
+                            LOG.warn("Unexpected field [{}] in event batch", field);
```

Comment:
For compatibility mostly, also according to our api guidelines unknown fields should be ignored.